Add spliting of peptides with python.

skrakau opened the pull request:
https://github.com/nf-core/epitopeprediction/issues/28

Ok, I did it in python in the end. With this max. 100 chunks will be created and only files with more than 5000 peptides will be split.That's what I would assume might be useful also for others. Let me know if you think this could be optimised.

#

Many thanks for contributing to nf-core/epitopeprediction!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

apeltzer commented:
General best practice would be to put this in a `foo.py` and place this script in the `/bin` folder of the pipeline and call it accordingly in the script section here, keeps the main.nf cleaner :-)

skrakau commented:
OK, will do! :)

christopher-mohr commented on main.nf:
```diff
@@ -38,6 +38,8 @@ def helpMessage() {
       --max_peptide_length [int]    Specifies the maximum peptide length Default: MHC class I: 11 aa, MHC class II: 16 aa 
       --min_peptide_length [int]    Specifies the minimum peptide length Default: MCH class I: 8 aa, MHC class II: 15 aa
       --tools [str]                 Specifies a list of tool(s) to use. Available are: 'syfpeithi', 'mhcflurry', 'mhcnuggets-class-1', 'mhcnuggets-class-2'. Can be combined in a list separated by comma.
+      --split_peptides_maxchunks [int]  Used in combination with '--peptides' or '--proteins': maximum number of peptide chunks that will be created for parallelization. Default: 100
```
How about `--peptides_split_maxchunks`and `peptides_split_chunksize`?

skrakau replied:
`split_peptides_minsize` is just the min. chunk size (apart for one file that contains the rest), it will become bigger if necessary to avoid more than `peptides_split_maxchunks` chunks .

`--peptides_split_minchunksize` or would that be too long?

skrakau replied:
You mean the actual number of chunks that is created? That would be a third parameter, but yes it might be a good solution to keep it simple. Or do you really not want to change the `*-minchunksize` parameter, then we could let the user define `*-maxchunks`, I just understood that you wanted to change that as well ...

christopher-mohr replied:
I mean that it might be sufficient to have a `maxchunks`parameter since that defines the size of the chunks anyway indirectly.

skrakau commented:
- changed `--split_peptides_maxchunks` -> `--peptides_split_maxchunks`
- changed `--split_peptides_minsize` -> `--peptides_split_minchunksize`
- improved help within `split_peptides.py` ("Max." -> "Maximum" etc.)
- adjusted whitespaces in epitopeprediction `helpMessage()`
